protoc_plugin/lib/src/enum_generator.dart

```diff
@@ -175,11 +174,27 @@ class EnumGenerator extends ProtobufContainer {
       out.println('];');
       out.println();
 
-      out.println(
-          'static final $coreImportPrefix.Map<$coreImportPrefix.int, $classname> _byValue ='
-          ' $protobufImportPrefix.ProtobufEnum.initByValue(values);');
-      out.println('static $classname? valueOf($coreImportPrefix.int value) =>'
-          ' _byValue[value];');
+      final useList = _canonicalValues.isEmpty ||
+          (_canonicalValues.every((val) => !val.number.isNegative) &&
+              _canonicalValues.length / (_canonicalValues.last.number + 1) >=
+                  0.7);
```

Comment:
Bug here: `_canonicalValues` is in the order the enum values appear in the source code, so the last enum here won't necessarily have the largest enum number.

I'll try to change order of this in g3 to see if it breaks anything. I don't think having `values` in source code order is useful. We should either sort it based on names (which will allow proto3 JSON decoder to binary search when decoding enums) or sort it based on values (so that the last element will give us the largest value both in runtime when initializing the lists, and in compile time).